homeassistant/components/lektrico/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""The Lektrico Charging Station integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from lektricowifi import lektricowifi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_FRIENDLY_NAME, CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+# List the platforms that you want to support.
+PLATFORMS = ["sensor", "binary_sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+PARALLEL_UPDATES = 1
+SCAN_INTERVAL = timedelta(seconds=10)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Lektrico Charging Station from a config entry."""
+    session = async_get_clientsession(hass)
+    charger = lektricowifi.Charger(
+        entry.data[CONF_HOST],
+        session=session,
+    )
+
+    # Ensure we can connect to it
+    try:
+        await charger.charger_info()
+    except lektricowifi.ChargerConnectionError as exception:
+        raise ConfigEntryNotReady("Unable to connect") from exception
```

Comment:
Let the update coordinator do this, it already calls charger_info on refresh.

Reply:
Ok, I deleted it.